### PR TITLE
DevEx: Destroy App Script

### PR DESCRIPTION
**Pull request by ericsorenson**

### CODE REVIEW

File: docs/ENVIRONMENT_DESTRUCTION.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Environment Destruction
+
+To destroy an environment that was previously created and deployed:
+
+1. In the AWS console, delete the DynamoDB tables for the environment in both us-east-1 and us-west-1. There should be two tables in us-east-1 and one table in us-west-1.
+
+2. Empty and delete the S3 buckets for the environment in both us-east-1 and us-west-1. There should be a total of 10 S3 buckets to empty and delete.
```
Comment (codyseibert) [inline docs/ENVIRONMENT_DESTRUCTION.md]: We might need to get more specific.  I think there is the deploy terraform bucket which probably shouldn't be cleared / deleted until AFTER terraform destroy.  Same with the dynamodb lock table maybe.

File: web-api/terraform/bin/environment-destroy.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+LOCK_TABLE=efcms-terraform-lock
+REGION=us-east-1
+
+rm -rf .terraform
+
+# export any outputs that serverless needs first from terraform
+# run serverless remove before running terraform destroy?
+
+terraform init -backend=true -backend-config=bucket="${BUCKET}" -backend-config=key="${KEY}" -backend-config=dynamodb_table="${LOCK_TABLE}" -backend-config=region="${REGION}"
+TF_VAR_my_s3_state_bucket="${BUCKET}" TF_VAR_my_s3_state_key="${KEY}" terraform destroy -auto-approve -var "dns_domain=${EFCMS_DOMAIN}" -var "environment=${ENVIRONMENT}" -var "cognito_suffix=${COGNITO_SUFFIX}" -var "ses_dmarc_rua=${SES_DMARC_EMAIL}" -var "cloudwatch_role_arn=${CLOUDWATCH_ROLE_ARN}" -var "post_confirmation_role_arn=${POST_CONFIRMATION_ROLE_ARN}" -var "es_instance_count=${ES_INSTANCE_COUNT}"
```
Comment (codyseibert) [inline web-api/terraform/bin/environment-destroy.sh]: We might want to document all these environment variables and what they should be set to before running the destroy.  Maybe it doesn't make a difference when doing a destroy though?

File: docs/ENVIRONMENT_DESTRUCTION.md (hunk 1 above)
Comment (rachelschneiderman) [reply docs/ENVIRONMENT_DESTRUCTION.md]: Is there a way to identify in the code which buckets should be removed? That way we don't have to 'hard code' the value 10 and risk deleting too much or not enough?

**Review by rachelschneiderman — APPROVED** [review]

File: web-api/terraform/bin/environment-destroy.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+BUCKET="${EFCMS_DOMAIN}.terraform.deploys"
+KEY="documents-${ENVIRONMENT}.tfstate"
+LOCK_TABLE=efcms-terraform-lock
+REGION=us-east-1
+
+rm -rf .terraform
+
+# export any outputs that serverless needs first from terraform
+# run serverless remove before running terraform destroy?
```
Comment (rachaelparris) [inline web-api/terraform/bin/environment-destroy.sh]: These comments can be removed. They're taken care of with manual steps.

File: docs/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -37,4 +37,24 @@ These libraries were forked to support multipart file uploads to s3 local.
...
+
+* aws_cloudfront_distribution.public_distribution: error creating CloudFront Distribution: InvalidViewerCertificate: The specified SSL certificate doesn't exist, isn't in us-east-1 region, isn't valid, or doesn't include a valid certificate chain.
+	status code: 400, request id: 88163d5d-bb9b-43db-abd7-57ba923cb103
+* module.environment.aws_cloudfront_distribution.distribution: 1 error(s) occurred:
+
+* aws_cloudfront_distribution.distribution: error creating CloudFront Distribution: InvalidViewerCertificate: The specified SSL certificate doesn't exist, isn't in us-east-1 region, isn't valid, or doesn't include a valid certificate chain.
+	status code: 400, request id: 8fb7c31a-8e7a-4608-ac7b-10d118deae59
+```
+
+If the occurs, rerun the build.
```
Comment (rachaelparris) [inline docs/TROUBLESHOOTING.md]: the => this

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**